actions/generate-k6-manifests/cmd/generator.go

```diff
@@ -0,0 +1,307 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"maps"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	yaml "gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Generator interface {
+	Initialize(filePath string) *ConfigFile
+	Generate()
+	HandleConfigFile(defaultConfigFile string, testType string) map[string]interface{}
+	HandleConfigFileOverride(base map[string]interface{}, overrideConfigFile string) map[string]interface{}
+	CallK6Archive(uniqName string, testConfigFileToUse string, testFile string)
+	CallKubectl(uniqName string, namespace string)
+	CallJsonnet(uniqName string, namespace string, environment string, parallelism int, nodeType string, sealedSecretName string, extraEnvVars []byte, resources []byte)
```

Comment:
Are any of the other functions used by callers outside this package? If so should the interface only expose that function to make it more generic?

Reply:
The main reason I created the interface was to have a simple way to reference the steps I needed to do  to generate the manifest files. I don't think the Interface has much use outside of that.

Reply:
So the interface is mainly a docs thing not functional?

Reply:
Pretty much yea.